GMG/GMG/Core/Services/PlaybackManager.swift

```diff
@@ -0,0 +1,67 @@
+//  Copyright © 2025 ADA 4th GMG. All rights reserved.
+
+import AudioKit
+import Combine
+import Foundation
+
+@Observable
+final class PlaybackManager {
+    private var audioPlayer: AudioPlayer
+
+    private(set) var isPlayingPublisher: CurrentValueSubject<Bool, Never>
+
+    private var isPlayingPublisherTimer: AnyCancellable?
+
+    init() {
+        self.audioPlayer = AudioPlayer()
+
+        self.isPlayingPublisher = CurrentValueSubject<Bool, Never>(false)
+
+        self.isPlayingPublisherTimer = nil
+    }
+
+    func play(_ url: URL) throws {
+        try AudioConductor.shared.setAudioMode(.playback)
+        AudioConductor.shared.addOutput(audioPlayer)
+        try AudioConductor.shared.start()
+
+        self.isPlayingPublisherTimer = Timer.publish(
+            every: 0.1,
+            on: RunLoop.main,
+            in: RunLoop.Mode.default
+        )
+        .autoconnect()
+        .sink { [weak self] _ in
+            guard let self else { return }
+
+            let isPlaying: Bool = self.audioPlayer.isPlaying
+
+            self.isPlayingPublisher.send(isPlaying)
+
+            if self.audioPlayer.currentTime == self.audioPlayer.duration {
+                Task {
+                    self.stop()
+                }
+            }
+        }
+
+        try self.audioPlayer.load(url: url)
```

Comment:
이 부분에서 load가 throw되면 위에 엔진과 세션이 열린 상태로 플레이어만 실패한 형태이게 되는데 이 때 오디오 세션이 활성화 된 상태여서 문제가 생길 수 있을 것 같습니다. 그래서 전체적으로 try-catch로 변경하면 어떨까 싶습니다!
```suggestion
        do {
                try self.audioPlayer.load(url: url)
                self.audioPlayer.play()
                self.audioPlayer.isLooping = false
                self.isPlayingPublisher.send(true)
         }  catch  {
                 stop()
                 throw error
```